packages/amplify-category-api/src/commands/api/add-graphql-datasource.js

```diff
@@ -206,3 +211,35 @@ async function getAwsClient(context, action) {
   const provider = require(providerPlugins[providerName]);
   return await provider.getConfiguredAWSClient(context, 'aurora-serverless', action);
 }
+
+function readSchema(projectDirectory) {
+  const schemaFilePath = path.join(projectDirectory, 'schema.graphql');
+  const schemaDirectoryPath = path.join(projectDirectory, 'schema');
+  const schemaFileExists = fs.existsSync(schemaFilePath);
+  const schemaDirectoryExists = fs.existsSync(schemaDirectoryPath);
+  let schema;
+  if (schemaFileExists) {
+    schema = fs.readFileSync(schemaFilePath).toString();
+  } else if (schemaDirectoryExists) {
+    schema = readSchemaDocuments(schemaDirectoryPath).join('\n');
+  } else {
+    throw new Error(`Could not find a schema at ${schemaFilePath}`);
+  }
+  return schema;
+}
+
+function readSchemaDocuments(schemaDirectoryPath) {
+  const files = glob.sync(schemaDirectoryPath + '/**/*');
+  let schemaDocuments = [];
+  for (const file of files) {
+    if (file.indexOf('.') === 0) {
+      continue;
+    }
+    const stats = fs.lstatSync(file);
+    if (stats.isFile() && file.split('.')[file.split('.').length - 1] === 'graphql') {
```

Comment:
glob only returns files matching the pattern and no directory is returned. With the updated glob you dont need this check